tidy3d/components/simulation.py

```diff
@@ -636,6 +677,46 @@ def background_structure(self) -> Structure:
         geometry = Box(size=(inf, inf, inf))
         return Structure(geometry=geometry, medium=self.medium)
 
+    @staticmethod
+    def intersecting_media(
```

Comment:
could consider making this a method of `Box`? so it could be used to find media intersecting with `self`. Would there be issues associated with this? I guess for one, we'd need to move `filter_structures_plane` to box as well? just thinking out loud if this makes this (potentially useful) method more accessible or just complicates things.

Reply:
The only other obstacle I see is that `Structure` type hints in this function and in `_filter_structures_plane` would require importing `structure`, which would be a circular import. Also, `_filter_structures_plane` would have to become a public method. `_filter_structures_plane` also has a `Medium` type hint, which may also cause circularity.